Fix for nested param/typeparam suggestions

lorcanmooney opened the pull request:
Fixes #11487.

This separates out the top-level `param`/`typeparam`-element suggestions from `name`-attribute suggestions, and calls each in the correct context. It also removes a couple of redundant symbol lookups in the VB implementation.

Pilchie commented:
Tag @dotnet/roslyn-ide 

test vsi please

tannergooding commented:
test windows_eta_open_prtest please

lorcanmooney commented:
test windows_vsi_p3_open_prtest please

crummel commented:
Build triggered. sha1 is original commit.

Pilchie commented on src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/XmlDocCommentCompletionProviderTests.vb:
```diff
@@ -293,6 +293,50 @@ End Class
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function TestNoNestedParam() As Task
```
You could add the `<WorkItem>` attributes here too.

Pilchie commented:
This LGTM - anyone on @dotnet/roslyn-ide want to review?  @CyrusNajmabadi and/or @rchande  in particular.

Pilchie commented:
@lorcanmooney any chance you could bring this up to date?

lorcanmooney commented:
@Pilchie Done, and added the missing work item attributes.

There are a few extra changes in there. Someone removed a lot of the completion spans threaded through these providers, but didn't quite finish the job. Since there were a bunch of merge conflicts around that, I removed whatever was left.

lorcanmooney commented:
Would anyone else like to take a look at this?

CyrusNajmabadi commented:
I will look

CyrusNajmabadi commented:
:+1:

Pilchie commented:
Just because it's been a few weeks, @dotnet-bot retest this please

Pilchie commented:
and test vsi please

lorcanmooney commented:
Just the usual reminder that I don't have access to the vsi results. If there's a failure I need to investigate, please let me know here.

Pilchie commented:
retest windows_vsi_p1_open_prtest please
retest windows_vsi_p2_open_prtest please 

p1 seems to have a hung machine, and in p2 the test automation failed to connect to VS.

lorcanmooney commented:
Can someone take a look at vsi_p2 please? I'd _really_ like to get this in so I can finish up some additional XML doc fixes and features.

Pilchie commented:
The failure is in an Extract Method test, where it's expecting to see two extract method options but only found one (from the screenshot).  @balajikris @heejaechang any idea what might have happened here?  I highly doubt it's related to @lorcanmooney's changes.

balajikris commented:
I suspect the test is flakey here, couldn't imagine this change affecting ExtractMethod feature. though I haven't seen this test fail this way before :)

Pilchie commented:
retest windows_vsi_p2_open_prtest please - unrelated extract method failure at http://dotnet-ci.cloudapp.net/job/Private/job/dotnet_roslyn-internal/job/master/job/windows_vsi_p2_open_prtest/458/

heejaechang commented:
@Pilchie looks like some race on option setting? looks like setExtractMethodOption right before the test didn't actually observed by extract method refactoring so it returns just 1.

Pilchie commented:
Tag @jasonmalinowski - any idea how @heejaechang's idea might have happened?

lorcanmooney commented:
Now that all the tests are green, I presume this is good to merge?

Pilchie commented:
Done - thanks again for the contribution, and the ping.

jasonmalinowski commented:
@heejaechang: when options change, what code invalidates previously computed analyzers or code fixes or whatever? I guess we have a longstanding bug/design issue where if you have a Solution snapshot the only way to get options is to jump back to the workspace where you might be seeing options from "the future". This would seem like something didn't invalidate the computation-already-in-flight after it was changed.
